x/eibc/keeper/msg_server.go

```diff
@@ -66,3 +69,66 @@ func (m msgServer) FulfillOrder(goCtx context.Context, msg *types.MsgFulfillOrde
 
 	return &types.MsgFulfillOrderResponse{}, err
 }
+
+// UpdateDemandOrder implements types.MsgServer.
+func (m msgServer) UpdateDemandOrder(goCtx context.Context, msg *types.MsgUpdateDemandOrder) (*types.MsgUpdateDemandOrderResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	err := msg.ValidateBasic()
+	if err != nil {
+		return nil, err
+	}
+
+	// Check that the order exists in status PENDING
+	demandOrder, err := m.GetActiveOrder(ctx, msg.OrderId)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check that the submitter is the expected recipient of the order
+	submitter := msg.GetSubmitterAddr()
+	if !submitter.Equals(demandOrder.GetRecipientBech32Address()) {
+		return nil, errorsmod.Wrap(sdkerrors.ErrUnauthorized, "only the recipient can update the order")
+	}
+
+	raPacket, err := m.dack.GetRollappPacket(ctx, demandOrder.TrackingPacketKey)
+	if err != nil {
+		return nil, err
+	}
+
+	var data transfertypes.FungibleTokenPacketData
+	if err := transfertypes.ModuleCdc.UnmarshalJSON(raPacket.GetPacket().GetData(), &data); err != nil {
+		return nil, err
+	}
+
+	// calculate the new price: transferTotal - newFee - bridgingFee
+	newFeeInt, _ := sdk.NewIntFromString(msg.NewFee)
+	transferTotal, _ := sdk.NewIntFromString(data.Amount)
+	newPrice := transferTotal.Sub(newFeeInt)
+	if newPrice.IsNegative() {
+		return nil, types.ErrFeeTooHigh
+	}
+	bridgingFee := m.dack.BridgingFeeFromAmt(ctx, transferTotal)
+	newPrice = newPrice.Sub(bridgingFee)
+
+	denom := demandOrder.Price[0].Denom
+	demandOrder.Fee = sdk.NewCoins(sdk.NewCoin(denom, newFeeInt))
+	demandOrder.Price = sdk.NewCoins(sdk.NewCoin(denom, newPrice))
```

Comment:
so much duplicated computation
please dedup/dry out with the other similar computations on the order creation etc